### PR TITLE
Use baselines for inlay hints testing

### PR DESCRIPTION
**Pull request by MariaSolOs**
Will make the tests in #54734 easier to update (and review the changes).

### CODE REVIEW

File: src/harness/fourslashImpl.ts [hunk 1]
```diff
@@ -822,6 +822,30 @@ export class TestState {
...
+            annotation += [
+                `text: "${hint.text}"`,
+                `position: ${hint.position}`,
+                `kind: ${hint.kind}`,
+                `whitespaceBefore: ${hint.whitespaceBefore}`,
+                `whitespaceAfter: ${hint.whitespaceAfter}`,
+            ].join(",\n\t");
+            annotation += ",\n}\n";
+            return annotation;
+        });
```
Comment (DanielRosenwasser) [inline src/harness/fourslashImpl.ts:844]: Any reason you can't just do

```suggestion
        annotations = JSON.stringify(hints.sort(sortHints));
```

**Comment by DanielRosenwasser** [comment]
This is a great start. One thing I think that gets missed in baselines is that they really should be somewhat human-readable.

So in addition to the raw JSON of every response, I think you should strive to give a preview of the inlay hint text. For example, let's say you have a file like

```ts
let x = y;
```

It would be desirable to have something like the following:

```ts
let x: number = y
     ^^^^^^^^
{
    text: ": number",
    position: 5,
    kind: Variable,
    ...
}
```

File: src/harness/fourslashImpl.ts (hunk 1 above)
Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts:844]: Because I want to format stuff lol

File: src/harness/fourslashImpl.ts [hunk 2]
```diff
@@ -822,6 +822,27 @@ export class TestState {
...
+        const baselineFile = this.getBaselineFileNameForContainingTestFile();
+        const fileName = this.activeFile.fileName;
+        const hints = this.languageService.provideInlayHints(fileName, span, preferences);
+        const annotations = ts.map(hints.sort(sortHints), hint => {
+            const span = { start: hint.position, length: hint.text.length };
+            const startLc = this.languageServiceAdapterHost.positionToLineAndCharacter(fileName, span.start);
+            const underline = " ".repeat(startLc.character) + "^";
+            let annotation = this.getFileContent(fileName).split(/\r?\n/)[startLc.line];
+            annotation += "\n" + underline + "\n" + JSON.stringify(hint, undefined, "  ");
+            return annotation;
```
Comment (DanielRosenwasser) [inline src/harness/fourslashImpl.ts:840]: Kind of wasteful. Here's how we do it based on reporting diagnostics:

```suggestion
            const sourceText = this.getFileContent(fileName);
            const lastLineIndex = getLineAndCharacterOfPosition(file, file.text.length).line
            const endPosition = startLc.line < lastLineIndex ? getPositionOfLineAndCharacter(file, i + 1, 0) : file.text.length;
            let lineContent = file.text.slice(, lineEnd);
            const annotation = lineContent.trimEnd() + "\n" + underline + "\n" + JSON.stringify(hint, undefined, "  ");
```

File: src/harness/fourslashImpl.ts [hunk 3]
```diff
@@ -822,6 +822,27 @@ export class TestState {
...
+        const sortHints = (a: HasPosition, b: HasPosition) => {
+            return a.position - b.position;
+        };
+
+        const baselineFile = this.getBaselineFileNameForContainingTestFile();
+        const fileName = this.activeFile.fileName;
+        const hints = this.languageService.provideInlayHints(fileName, span, preferences);
+        const annotations = ts.map(hints.sort(sortHints), hint => {
+            const span = { start: hint.position, length: hint.text.length };
+            const startLc = this.languageServiceAdapterHost.positionToLineAndCharacter(fileName, span.start);
```
Comment (DanielRosenwasser) [inline src/harness/fourslashImpl.ts:836]: you only use `startLc` for the line, so I would just extract the line out immediately

Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts:836]: I use the character for creating the underline.

File: src/harness/fourslashImpl.ts (hunk 2 above)
Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts:840]: What is the `SourceFileLike` parameter for `getLineAndCharacterOfPosition` that I would use here?

File: src/harness/fourslashImpl.ts (hunk 3 above)
Comment (DanielRosenwasser) [reply src/harness/fourslashImpl.ts:836]: Duh!

File: src/harness/fourslashImpl.ts (hunk 2 above)
Comment (DanielRosenwasser) [reply src/harness/fourslashImpl.ts:840]: Would `this.activeFile` work?

Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts]: nope

Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts]: or well yeah I can adjust the object for the call

Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts]: Although idk I prefer to leave it as is.

**Comment by MariaSolOs** [comment]
@weswigham Could you please take a look?

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**